File: src/api/helpers/timings.ts
import type { SessionHandle } from "~/models";

export const translatePositionToTimings = (
  { instance: { endings } }: SessionHandle,
  position: number
): { hours: number, minutes: number } => {
  if (position > endings.length) {
    position = position % (endings.length - 1);
  }

  const formatted = endings[position];

  const [hours, minutes] = formatted.split("h").map(Number);
  return { hours, minutes };
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/helpers/timings.kt
package ink.literate.pawnote.api.helpers

import ink.literate.pawnote.models.InstanceParameters

data class Timing(val hours: Short, val minutes: Short)

fun translatePositionToTimings(sessionInstance: InstanceParameters, position: Int): Timing {
  var finalPos = position

  if (finalPos > sessionInstance.endings.count()) finalPos %= (sessionInstance.endings.count() - 1)

  val formatted = sessionInstance.endings[position]

  val (hours, minutes) = formatted.split('h').map { it.toShort() }

  return Timing(hours, minutes)
}
